In [1]:
from scenedetect import detect, ContentDetector, split_video_ffmpeg, SceneManager, AdaptiveDetector, ThresholdDetector, HashDetector, HistogramDetector, StatsManager
video_path = '/Users/spectatr/Downloads/soccernet_data/england_epl/2014-2015/2015-02-21 - 18-00 Chelsea 1 - 1 Burnley/2_720p.mkv'
scene_list = detect(video_path, ContentDetector()) #thresholod inside content detection: COntent got 249 frames.
scene_list # 

[(00:00:00.000 [frame=0, fps=25.000], 00:00:33.720 [frame=843, fps=25.000]),
 (00:00:33.720 [frame=843, fps=25.000], 00:00:37.600 [frame=940, fps=25.000]),
 (00:00:37.600 [frame=940, fps=25.000], 00:00:43.680 [frame=1092, fps=25.000]),
 (00:00:43.680 [frame=1092, fps=25.000],
  00:01:06.720 [frame=1668, fps=25.000]),
 (00:01:06.720 [frame=1668, fps=25.000],
  00:01:13.680 [frame=1842, fps=25.000]),
 (00:01:13.680 [frame=1842, fps=25.000],
  00:01:43.520 [frame=2588, fps=25.000]),
 (00:01:43.520 [frame=2588, fps=25.000],
  00:01:44.840 [frame=2621, fps=25.000]),
 (00:01:44.840 [frame=2621, fps=25.000],
  00:01:48.960 [frame=2724, fps=25.000]),
 (00:01:48.960 [frame=2724, fps=25.000],
  00:02:00.840 [frame=3021, fps=25.000]),
 (00:02:00.840 [frame=3021, fps=25.000],
  00:02:05.640 [frame=3141, fps=25.000]),
 (00:02:05.640 [frame=3141, fps=25.000],
  00:02:10.160 [frame=3254, fps=25.000]),
 (00:02:10.160 [frame=3254, fps=25.000],
  00:02:16.640 [frame=3416, fps=25.000]),
 (00:02:16.640 [f

In [6]:
from scenedetect import detect, ContentDetector, split_video_ffmpeg, SceneManager, AdaptiveDetector, ThresholdDetector, HashDetector, HistogramDetector, StatsManager
video_path = '/Users/spectatr/Downloads/soccernet_data/england_epl/2014-2015/2015-02-21 - 18-00 Chelsea 1 - 1 Burnley/2_720p.mkv'
scene_list = detect(video_path, HistogramDetector()) #thresholod inside content detection
scene_list
print(f"Lenght of scene list is {len(scene_list)}") 

Lenght of scene list is 318


# 
- custom our algorithm took 20 minutes to extract framechange with 209 frames.
- Lenght is 307 for HashDetector took 2minutes.
- Scene Detector is 153 frames in 30 seconds
- content dectector took 2 minutes for 270 frames
- Adaptive only created 152 = wrong
- Histogram detector took = 2 minuste with 318

In [2]:
for i, scene in enumerate(scene_list):
    print('    Scene %2d: Start %s / Frame %d, End %s / Frame %d' % (
        i+1,
        scene[0].get_timecode(), scene[0].get_frames(),
        scene[1].get_timecode(), scene[1].get_frames(),))


    Scene  1: Start 00:00:00.000 / Frame 0, End 00:00:33.720 / Frame 843
    Scene  2: Start 00:00:33.720 / Frame 843, End 00:00:37.600 / Frame 940
    Scene  3: Start 00:00:37.600 / Frame 940, End 00:00:43.680 / Frame 1092
    Scene  4: Start 00:00:43.680 / Frame 1092, End 00:01:06.720 / Frame 1668
    Scene  5: Start 00:01:06.720 / Frame 1668, End 00:01:13.680 / Frame 1842
    Scene  6: Start 00:01:13.680 / Frame 1842, End 00:01:43.520 / Frame 2588
    Scene  7: Start 00:01:43.520 / Frame 2588, End 00:01:44.840 / Frame 2621
    Scene  8: Start 00:01:44.840 / Frame 2621, End 00:01:48.960 / Frame 2724
    Scene  9: Start 00:01:48.960 / Frame 2724, End 00:02:00.840 / Frame 3021
    Scene 10: Start 00:02:00.840 / Frame 3021, End 00:02:05.640 / Frame 3141
    Scene 11: Start 00:02:05.640 / Frame 3141, End 00:02:10.160 / Frame 3254
    Scene 12: Start 00:02:10.160 / Frame 3254, End 00:02:16.640 / Frame 3416
    Scene 13: Start 00:02:16.640 / Frame 3416, End 00:02:51.600 / Frame 4290
    Sc

In [ ]:
split_video_ffmpeg(video_path, scene_list)